### DISCUSSION TITLE
About GumbelVectorQuantizer

### DISCUSSION THREAD

**Discussion by LYPinASR** [speechbrain/speechbrain#1877] [Q&A]
Hello! 

I am working on \recipes\LibriSpeech\self-supervised-learning\wav2vec2.

How can I get the IDs of all quantised outputs. In Class "GumbelVectorQuantizer", there are only target and meta. 

And how should I finetune a ASR model. Specifically, how can i get the parameters from a checkpoint.

Best!

**Comment by anautsch** [comment]
Hi @LYPinASR 

can you please describe a bit more?

If it is about the example IDs - would using the batch information do?

This line suggests the ID information should be available:
https://github.com/speechbrain/speechbrain/blob/beec862238c4c2ccfcb20d9ca592b671673ee6af/recipes/LibriSpeech/self-supervised-learning/wav2vec2/train_sb_wav2vec2.py#L292

In another recipe, the IDs are obtained this way:
https://github.com/speechbrain/speechbrain/blob/beec862238c4c2ccfcb20d9ca592b671673ee6af/recipes/LibriSpeech/ASR/CTC/train_with_wav2vec.py#L79

What confuses me a bit, this line should draw from `batch.sig` and not from `batch` (it could be a short-form syntax I'm unaware of)
https://github.com/speechbrain/speechbrain/blob/beec862238c4c2ccfcb20d9ca592b671673ee6af/recipes/LibriSpeech/self-supervised-learning/wav2vec2/train_sb_wav2vec2.py#L37-L41

Were you able to get these IDs or do you want to get other IDs?

---

About the fine-tuning, what do you want to fine-tune?

**Comment by anautsch** [comment]
Please take a look at the SB tutorials:
https://colab.research.google.com/drive/1LN7R3U3xneDgDRK2gC5MzGkLysCWxuC3?usp=sharing

... and some more:
https://speechbrain.github.io/tutorial_advanced.html
